### PR TITLE
SCALRCORE-24801 Integrations > Webhook update provider resource

### PR DESCRIPTION
**Pull request by petroprotsakh**

### CODE REVIEW

File: webhook_integration.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	IsShared *bool   `jsonapi:"attr,is-shared,omitempty"`
+
+	Url         *string          `jsonapi:"attr,url,omitempty"`
+	SecretKey   *string          `jsonapi:"attr,secret-key,omitempty"`
+	Timeout     *int             `jsonapi:"attr,timeout,omitempty"`
+	MaxAttempts *int             `jsonapi:"attr,max-attempts,omitempty"`
+	Headers     []*WebhookHeader `jsonapi:"attr,headers,omitempty"`
+
+	Environments []*Environment     `jsonapi:"relation,environments,omitempty"`
+	Events       []*EventDefinition `jsonapi:"relation,events,omitempty"`
```
Comment (DayS1eeper) [inline webhook_integration.go:99]: All options have `omitempty`.
Can `Environments` be set to `[]` with update request`? Can `Timeout` be set to 0 or some default value?

**Review by DayS1eeper — CHANGES_REQUESTED** [review]

File: webhook_integration.go (hunk 1 above)
Comment (petroprotsakh) [reply webhook_integration.go]: @DayS1eeper Thanks, I forgot about the `omitempty` issue with to-many relations in jsonapi lib. Fixed.
For `Timeout` and other attributes it's ok, they're pointers with zero value of `nil`

Comment (DayS1eeper) [reply webhook_integration.go]: @petroprotsakh 
>  For Timeout and other attributes it's ok, they're pointers with zero value of nil

https://pkg.go.dev/encoding/json#Marshal
> The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string.


With `omitempty` it is not possible to set `MaxAttempts=0` or `Timeout=0` or `Enabled=false` or `IsShared=false`.

Comment (DayS1eeper) [reply webhook_integration.go]: Please add tests for updating from `IsShared=true` to `IsShared=false`, `MaxAttempts=3` to `MaxAttempts=0`.

Comment (petroprotsakh) [reply webhook_integration.go]: @DayS1eeper https://github.com/Scalr/go-scalr/pull/105/files#diff-232db8d6b3359f96e57e7c98bfa6df3d2f05327ca7931898b372cd3c96375d65R122

Comment (petroprotsakh) [reply webhook_integration.go]: @DayS1eeper 
>With omitempty it is not possible to set MaxAttempts=0 or Timeout=0 or Enabled=false or IsShared=false.

The type of `Timeout`, `IsShared`, etc is a pointer, not int/bool. Zero value for pointer is nil, when set to `0`/`false` it's no more zero.

**Review by DayS1eeper — APPROVED** [review]

**Comment by emocharnik** [comment]
@petroprotsakh, Required builds are failing.

**Comment by emocharnik** [comment]
@petroprotsakh, Required builds are failing.

**Comment by emocharnik** [comment]
@petroprotsakh, Required builds are failing.

**Comment by emocharnik** [comment]
@petroprotsakh, Required builds are failing.

**Comment by emocharnik** [comment]
@petroprotsakh, Required builds are failing.

**Comment by emocharnik** [comment]
@petroprotsakh, Some non-required builds are failing or stuck.

**Pull request merged**